File: Qnerator/src/example/test_gen/Hello.cpp
#pragma pack(push, 1)
struct Hello {
    std::string message;

    Hello(std::string message) : message(message) {}

    std::vector<uint8_t> serialize() const {
        std::vector<uint8_t> buffer;
        uint32_t message_length = message.size();
        buffer.insert(buffer.end(), reinterpret_cast<const uint8_t*>(&message_length), reinterpret_cast<const uint8_t*>(&message_length + 1));
        buffer.insert(buffer.end(), message.begin(), message.end());
        return buffer;
    }

    static Hello deserialize(const std::vector<uint8_t>& buffer) {
        size_t offset = 0;
        uint32_t message_length = *reinterpret_cast<const uint32_t*>(buffer.data() + offset);
        offset += sizeof(uint32_t);
        std::string message(buffer.begin() + offset, buffer.begin() + offset + message_length);
        offset += message_length;
        return Hello(message);
    }
};
#pragma pack(pop)

File: Qnerator/src/example/test_gen/Hello.rs
// 자동 생성된 구조체 및 관련 메서드
        #[repr(C)]
        #[derive(Debug, Clone)]
        pub struct Hello {
            message: String,
    message_length: u32,}
    
        impl Hello {
            pub fn new(message: String) -> Self {
                Self {
                    message,        }
            }
    
            pub fn serialize(&self) -> Vec<u8> {
                let mut buffer = Vec::new();
                buffer.extend(&self.message.len().to_le_bytes());
buffer.extend(self.message.as_bytes());
                buffer
            }
    
            pub fn deserialize(buffer: &[u8]) -> io::Result<Self> {
                let mut offset = 0;
                let mut message_length_bytes = [0u8; 4];
message_length_bytes.copy_from_slice(&buffer[offset..offset + 4]);
let message_length = u32::from_le_bytes(message_length_bytes);
offset += 4;
let message = String::from_utf8(buffer[offset..offset + message_length as usize].to_vec())
.map_err(|_| io::Error::new(io::ErrorKind::InvalidData, "Invalid UTF-8 string"))?;
offset += message_length as usize;
                Ok(Self {
                                message,
                })
            }
        }